src/data/store.rs

```diff
@@ -94,64 +93,65 @@ impl Store {
 
             // If there is no chat, create an empty one
             if self.chats.get_current_chat().is_none() {
-                self.chats.create_empty_chat_with_model_file(file);
+                self.chats.create_empty_chat();
             }
         }
     }
 
     pub fn select_chat(&mut self, chat_id: ChatID) {
         self.chats.set_current_chat(chat_id);
+
+        if let Some(file_id) = self.get_last_used_file_id_in_current_chat() {
```

Comment:
nitpick: maybe we can extract this chunk of added code to a function called `self.load_last_model_for_current_chat()` and call it here. Just for the sake of having smaller and self documented functions.